metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutionMonitor.java

```diff
@@ -182,34 +185,38 @@ public void performFailsafe() {
    */
   public Pair<WorkflowExecution, Boolean> claimExecution(String workflowExecutionId) {
 
+    WorkflowExecution workflowExecution;
+    boolean claimed = false;
     try {
-
       // Lock for the duration of this request
       lock.lock();
 
       // Retrieve the most current version of the execution.
-      final WorkflowExecution workflowExecution = workflowExecutionDao.getById(workflowExecutionId);
-
-      final boolean claimed = mayClaimExecution(workflowExecution);
-      if (claimed) {
-        //Update dates
-        final Date now = new Date();
-        workflowExecution.setUpdatedDate(now);
-        if (workflowExecution.getWorkflowStatus() != WorkflowStatus.RUNNING) {
-          workflowExecution.setStartedDate(now);
-          workflowExecution.setWorkflowStatus(WorkflowStatus.RUNNING);
+      workflowExecution = workflowExecutionDao.getById(workflowExecutionId);
+
+      if (workflowExecution != null && CLAIMABLE_STATUSES
+          .contains(workflowExecution.getWorkflowStatus())) {
```

Comment:
I think you could leave out the second condition (the claimable statuses one)? It is the very first check in the method `mayClaimExecution` and there it actually prints a nice log statement.

Reply:
Okay, I'm changing it slightly though, so that the Set is used more explicitly